source/MaterialXContrib/MaterialXMaya/ShadingNodeOverrides.cpp

```diff
@@ -411,19 +428,34 @@ void MaterialXShadingNodeImpl<BASE>::updateShader(MHWRender::MShaderInstance& sh
                         samplerDescription.filter = filterModes[static_cast<int>(samplingProperties.filterType)];
                     }
 
-                    status = ::bindFileTexture(shader, textureParameterName, imageSearchPath, valueString,
+                    status = bindFileTexture(shader, textureParameterName, imageSearchPath, valueString,
                         samplerDescription, textureDescription, udimIdentifiers);
                 }
             }
         }
     }
 }
 
-namespace MHWRender
+template class ShadingNodeOverride<MHWRender::MPxShadingNodeOverride>;
+template class ShadingNodeOverride<MHWRender::MPxSurfaceShadingNodeOverride>;
+
+MHWRender::MPxSurfaceShadingNodeOverride*
+SurfaceOverride::creator(const MObject& obj)
 {
-    class MPxShadingNodeOverride;
-    class MPxSurfaceShadingNodeOverride;
-};
+    std::cout.rdbuf(std::cerr.rdbuf());
+    return new SurfaceOverride(obj);
+}
+
+MString
+SurfaceOverride::transparencyParameter() const
```

Comment:
Just curious if we want to put out alpha for textures do we also need the same mechanism as for surface shaders?

Reply:
This virtual is only exposed on MPxSurfaceShadingNodeOverride, so this mechanism is only applicable to surfaces. If this attribute is connected to anything at all in the DG (supposedly a texture's output transparency) then the surface is considered transparent.